fl4health/strategies/feddg_ga_strategy.py

```diff
@@ -0,0 +1,382 @@
+from enum import Enum
+from logging import WARNING
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+from flwr.common import MetricsAggregationFn, NDArrays, Parameters, ndarrays_to_parameters, parameters_to_ndarrays
+from flwr.common.logger import log
+from flwr.common.typing import EvaluateRes, FitIns, FitRes, Scalar
+from flwr.server.client_manager import ClientManager
+from flwr.server.client_proxy import ClientProxy
+from flwr.server.strategy import FedAvg
+
+from fl4health.client_managers.fixed_sampling_client_manager import FixedSamplingClientManager
+
+
+class FairnessMetricType(Enum):
+    """Defines the basic types for fairness metrics, their default names and their default signals"""
+
+    ACCURACY = "val - prediction - accuracy"
+    LOSS = "val - loss"
+    CUSTOM = "custom"
+
+    @classmethod
+    def signal_for_type(cls, fairness_metric_type: "FairnessMetricType") -> float:
+        """
+        Return the default signal for the given metric type.
+
+        Args:
+            fairness_metric_type: (FairnessMetricType) the fairness metric type.
+
+        Returns: (float) -1.0 if FairnessMetricType.ACCURACY, 1.0 if FairnessMetricType.LOSS and
+            0 if FairnessMetricType.CUSTOM.
+        """
+        if fairness_metric_type == FairnessMetricType.ACCURACY:
+            return -1.0
+        if fairness_metric_type == FairnessMetricType.LOSS:
+            return 1
+        return 0
+
+
+class FairnessMetric:
+    """Defines a fairness metric with attributes that can be overridden if needed."""
+
+    def __init__(
+        self,
+        metric_type: FairnessMetricType,
+        metric_name: str = "",
+        signal: float = 0.0,
+    ):
+        """
+        Instantiates a fairness metric with a type and optional metric name and
+            signal if one wants to override them.
+
+        Args:
+            metric_type: (FairnessMetricType) the fairness metric type. If CUSTOM, the metric_type and
+                signal should be provided.
+            metric_name: (str, optional) the name of the metric to be used as fairness metric.
+                Optional, default is metric_type.value.
+            signal: (float, optional) the signal of the fairness metric. Optional default is
+                FairnessMetricType.signal_for_type(metric_type)
+        """
+        self.metric_type = metric_type
+        self.metric_name = metric_name
+        self.signal = signal
+
+        if metric_type is FairnessMetricType.CUSTOM:
+            assert metric_name is not None and metric_name != "" and signal is not None
+        else:
+            if metric_name == "":
+                self.metric_name = metric_type.value
+            if signal == 0.0:
+                self.signal = FairnessMetricType.signal_for_type(metric_type)
+
+
+# Initial value for the adjustment weight of each client model
+INITIAL_ADJUSTMENT_WEIGHT = 1.0 / 3.0
+
+
+class FedDGGAStrategy(FedAvg):
```

Comment:
Very minor, but in other places, we've used the acronym naming convention of first letter uppercase, remainder lowercase (FENDA -> Fenda). This is a bit of a weird combo acronym. So I think our name would be `FedDgGaStrategy`? Sort of ugly....but follows the "rules"

Reply:
Yeah, pretty ugly :/ I'll do it but I prefer the other way even if it break the rules a bit.